Implementación de la vista del tablero

ZhangDreamer opened the pull request:
Este pull request agrega una nueva clase llamada _VistaTablero_ donde se agregan dos metodos, **mostrarTablero** y **obtenerSimboloCasilla** . 

El metodo **mostrarTablero** se encarga de imprimir el tablero de juego en la consola, mostrando las casillas y su estado correspondiente. Además, se ha agregado el metodo **obtenerSimboloCasilla** para obtener el símbolo que representa visualmente el estado de una casilla.

El método principal **mostrarTablero**, recibe un objeto _Tablero_ como argumento. Dentro de este método, se obtiene la matriz de objetos _Casilla_ del tablero utilizando el método **getCasillas**.

```
     public static void mostrarTablero(Tablero tablero) {
         Casilla[][] casillas = tablero.getCasillas();
```

A continuación, el código muestra los encabezados de las columnas utilizando un bucle for. Imprime los números del 1 al número de columnas de la matriz casillas[0] (que deberian ser hasta 9) para representar las columnas del tablero.
```
         System.out.print("  ");
          for (int i = 0; i < casillas[0].length; i++) {
            System.out.print(i + 1 + " ");
         }
```


Se utiliza otro bucle for para iterar a través de las filas de la matriz casillas. Para cada fila, imprime una letra como encabezado de la fila (comenzando desde 'A' y aumentando por cada fila) y luego recorre cada columna en esa fila.
Se crea un objeto _Coordenada_ con la letra de la fila y el número de columna correspondientes. Con esto se puede obtener la casilla especifica del tablero para imprimirse.
```
 for (int i = 0; i < casillas.length; i++) {
            System.out.print((char) ('A' + i) + " ");
            for (int j = 0; j < casillas[i].length; j++) {
                Coordenada coordenada = new Coordenada((char) ('A' + i), j + 1);
                Casilla casilla = tablero.getCasilla(coordenada);
                System.out.print(obtenerSimboloCasilla(casilla) + " ");
            }
            System.out.println();
        }
    }
```

Se llama al método obtenerSimboloCasilla para obtener el símbolo que representa el estado de la casilla. 

```
    private static char obtenerSimboloCasilla(Casilla casilla) {
        switch (casilla.getEstado()) {
            case VACIA:
                return '-';
            case OCUPADA:
                return '▉';
            case ATACADA:
                return 'X';
            default:
                return '?';
        }
    }
```

Hay que notar que no hay un estado en la casilla donde representa un ataque fallado ○, sugiero agregarlo como estado en la casilla pero tambien pueden sugerir otra resolucion.

ZhangDreamer committed:
Merge pull request #12 from Skiper132/main

merge main to Anyel

ZhangDreamer committed:
Merge branch 'main' of https://github.com/Skiper132/battleship-java into Anyel

ZhangDreamer committed:
Merge pull request #20 from Skiper132/main

merge with latest update on main

ZhangDreamer committed:
Merge pull request #24 from Skiper132/main

merge with main

ZhangDreamer committed:
Added Vista Tablero

Skiper132 commented:
En lugar de crear una nueva clase VistaTablero, sugiero que incorporemos los métodos mostrarTablero y obtenerSimboloCasilla directamente en la clase Tablero. De esta manera, seguimos los principios de la programación orientada a objetos y mantenemos nuestro código más limpio y conciso.

En lugar de duplicar la responsabilidad de determinar el símbolo de una casilla entre la clase Tablero y las enumeraciones ResultadoAtaque y EstadoCasilla, podríamos mover esta responsabilidad directamente a la clase Casilla. Para ello, podemos añadir un atributo simbolo en Casilla que será actualizado en función del resultado de un ataque.

Con este approach modificamos el método atacarCasilla() un poco para que también actualice el símbolo de la casilla de acuerdo con el resultado del ataque. Aquí te dejo un ejemplo de cómo podríamos implementar esto:

java
Copy code
public class Casilla {
    private EstadoCasilla estado;
    private char simbolo;

    public Casilla() {
        this.estado = EstadoCasilla.VACIA;
        this.simbolo = estado.getSimbolo();
    }

    public void setEstado(EstadoCasilla estado) {
        this.estado = estado;
        this.simbolo = estado.getSimbolo();
    }

    public char getSimbolo() {
        return simbolo;
    }

    public ResultadoAtaque atacarCasilla() throws CasillaYaAtacadaException {
        EstadoCasilla estadoActual = this.estado;
        if (estadoActual == EstadoCasilla.ATACADA) {
            throw new CasillaYaAtacadaException();
        }
        this.estado = EstadoCasilla.ATACADA;
        this.simbolo = estado.getSimbolo(); // actualizamos el símbolo de acuerdo al nuevo estado

        if (estadoActual == EstadoCasilla.OCUPADA) {
            return ResultadoAtaque.ACIERTO;
        } else {
            this.simbolo = '○'; // símbolo para ataque fallido
            return ResultadoAtaque.FALLA;
        }
    }

    // ...resto de la clase Casilla...
}


Dado a que me dices que no estarás disponible hasta la noche cerraré el PR y agregaré las correcciones mencionadas.

aldoapicella committed:
Changes to be committed:
	new file:   src/com/battleship/controlador/ControladorJuego.java
	modified:   src/com/battleship/modelo/Tablero.java
	modified:   src/com/battleship/vista/Pantalla.java
 Changes not staged for commit:
	modified:   bin/com/battleship/controlador/Controlador.class
	modified:   bin/com/battleship/modelo/Barco.class
	modified:   bin/com/battleship/modelo/Casilla.class
	modified:   bin/com/battleship/modelo/Direccion.class
	modified:   bin/com/battleship/modelo/EstadoCasilla.class
	modified:   bin/com/battleship/modelo/Jugador.class
	modified:   bin/com/battleship/modelo/Tablero.class
	modified:   bin/com/battleship/vista/Pantalla.class
	modified:   src/com/battleship/modelo/Casilla.java
	modified:   src/com/battleship/modelo/EstadoCasilla.java
 Untracked files:
	bin/com/battleship/controlador/ControladorJuego.class
	bin/com/battleship/modelo/NombreBarco.class
	bin/com/battleship/vista/VistaTablero.class

aldoapicella committed:
modified:   src/com/battleship/controlador/Controlador.java
	modified:   bin/com/battleship/controlador/Controlador.class
	modified:   bin/com/battleship/modelo/Barco.class
	modified:   bin/com/battleship/modelo/Casilla.class
	modified:   bin/com/battleship/modelo/Direccion.class
	modified:   bin/com/battleship/modelo/EstadoCasilla.class
	modified:   bin/com/battleship/modelo/Jugador.class
	modified:   bin/com/battleship/modelo/Tablero.class
	modified:   bin/com/battleship/vista/Pantalla.class
	deleted:    src/com/battleship/controlador/ControladorJuego.java
	modified:   src/com/battleship/modelo/Casilla.java
	modified:   src/com/battleship/modelo/EstadoCasilla.java
 Untracked files:
	bin/com/battleship/modelo/NombreBarco.class
	bin/com/battleship/vista/VistaTablero.class

aldoapicella committed:
modified:   bin/com/battleship/controlador/Controlador.class
	modified:   bin/com/battleship/modelo/Barco.class
	modified:   bin/com/battleship/modelo/Casilla.class
	modified:   bin/com/battleship/modelo/Direccion.class
	modified:   bin/com/battleship/modelo/EstadoCasilla.class
	modified:   bin/com/battleship/modelo/Jugador.class
	new file:   bin/com/battleship/modelo/NombreBarco.class
	modified:   bin/com/battleship/modelo/Tablero.class
	modified:   bin/com/battleship/vista/Pantalla.class
	new file:   bin/com/battleship/vista/VistaTablero.class
	deleted:    src/com/battleship/controlador/ControladorJuego.java
	modified:   src/com/battleship/modelo/Casilla.java
	modified:   src/com/battleship/modelo/EstadoCasilla.java
	modified:   src/com/battleship/modelo/Jugador.java

aldoapicella committed:
Changes to be committed:
	modified:   bin/com/battleship/controlador/Controlador.class
	modified:   bin/com/battleship/modelo/Barco.class
	modified:   bin/com/battleship/modelo/Jugador.class
	modified:   bin/com/battleship/modelo/ResultadoAtaque.class
	modified:   src/com/battleship/controlador/Controlador.java
	modified:   src/com/battleship/modelo/Barco.java
	modified:   src/com/battleship/modelo/ResultadoAtaque.java
